scm-webapp/src/main/java/sonia/scm/api/v2/resources/IndexDtoGenerator.java

```diff
@@ -120,4 +121,11 @@ public IndexDto generate() {
 
     return new IndexDto(builder.build(), embeddedBuilder.build(), scmContextProvider.getVersion());
   }
+
+  private boolean shouldAppendSubjectRelatedLinks() {
+    return (SecurityUtils.getSubject().isAuthenticated()
+      && !Authentications.isAuthenticatedSubjectAnonymous())
+      || (Authentications.isAuthenticatedSubjectAnonymous()
+      && configuration.getAnonymousMode() == AnonymousMode.FULL);
+  }
```

Comment:
please break the method into smaller understandable parts, because it is very hard to read.